### PR TITLE
changes to get Trilinos building on crusher

### PR DESCRIPTION
**Pull request by tasmith4**
Should be the last of the changes from #352.

### CODE REVIEW

File: repos/exawind/packages/trilinos/package.py [hunk 1]
```diff
@@ -23,7 +23,6 @@ class Trilinos(bTrilinos):
     variant("asan", default=False,
             description="Turn on address sanitizer")
 
-    patch("kokkos_zero_length_team.patch")
```
Comment (psakievich) [inline repos/exawind/packages/trilinos/package.py:26]: We should keep this patch and do:
```python
patch("kokkos_zero_length_team.patch", when'@:13.3.0')
```
to support the old version of Trilinos until we remove it.

**Review by psakievich — CHANGES_REQUESTED** [review]

File: repos/exawind/packages/trilinos/package.py [hunk 2]
```diff
@@ -23,7 +23,6 @@ class Trilinos(bTrilinos):
     variant("asan", default=False,
             description="Turn on address sanitizer")
```
Comment (tasmith4) [inline repos/exawind/packages/trilinos/package.py:25]:
```suggestion

patch("kokkos_zero_length_team.patch", when'@:13.3.0')

```

File: repos/exawind/packages/trilinos/package.py [hunk 3]
```diff
@@ -23,7 +23,8 @@ class Trilinos(bTrilinos):
...
 
-    patch("kokkos_zero_length_team.patch")
+patch("kokkos_zero_length_team.patch", when'@:13.3.0')
```
Comment (tasmith4) [inline repos/exawind/packages/trilinos/package.py]:
```suggestion
    patch("kokkos_zero_length_team.patch", when'@:13.3.0')
```

File: repos/exawind/packages/trilinos/package.py (hunk 1 above)
Comment (tasmith4) [reply repos/exawind/packages/trilinos/package.py:26]: ok how about that?

**Review by psakievich — APPROVED** [review]

File: repos/exawind/packages/trilinos/package.py [hunk 4]
```diff
@@ -23,7 +23,8 @@ class Trilinos(bTrilinos):
     variant("asan", default=False,
             description="Turn on address sanitizer")
 
-    patch("kokkos_zero_length_team.patch")
+    patch("kokkos_zero_length_team.patch", when'@:13.3.0')
```
Comment (tasmith4) [inline repos/exawind/packages/trilinos/package.py]:
```suggestion
    patch("kokkos_zero_length_team.patch", when="@:13.3.0")
```

**Review by psakievich — APPROVED** [review]

**Pull request merged**